src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -1492,10 +1493,14 @@ def item(self) -> None:
         ValueError
             If the data is not length = 1.
         """
-        # TODO: SNOW-1458117 implement item
+        if len(self) == 1:
+            return self.tolist()[0]
```

Comment:
No this is not efficient which will pull the whole index locally.

Reply:
yep but the whole index is only one value in this case, that should be fine for us right?

Reply:
This API is a bit weird, it is only supposed to work when the length of the index is 1. Native pandas also raises a value error when the index length is not 1

Reply:
It should not be too bad because length is a cached property? Maybe we can discuss this more